src/test/java/com/dsrm/dsrmbackend/UserControllerIntegrationTests.java

```diff
@@ -87,4 +99,43 @@ void retrieveUsersWithoutRoles() throws Exception {
 
     }
 
+
+    @Test
+    void validPatchExistingUserReturnStatus() throws Exception {
+        ObjectMapper objectMapper = new ObjectMapper();
+        UserRolesOnlyDTO userRolesOnlyDTO  = new UserRolesOnlyDTO();
+        List<Long> longs = new ArrayList<>();
+        longs.add(1L);
+        userRolesOnlyDTO.setRoles(longs);
+        this.mockMvc.perform(patch("/users/1/roles").contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(userRolesOnlyDTO))
+                ).andExpect(status().isOk());
+
+    }
+
+    @Test
+    void validPatchExistingUser(){
+        UserRolesOnlyDTO userRolesOnlyDTO  = new UserRolesOnlyDTO();
+        List<Long> longs = new ArrayList<>();
+        longs.add(3L);
+        userRolesOnlyDTO.setRoles(longs);
+        Optional<User> userOptional = userService.updateUser(userRolesOnlyDTO,1L);
```

Comment:
It doesn't test anything. Please use mockmvc and after updating user, retrieve it by id from repository and check if role were added and if tit has right role.

Reply:
I made the test result dependent on UserRepository, but I don't see the use of mockmvc at first glance. From what I searched I saw testing using when, but I don't think it's necessary here, if I'm wrong I'll try to correct it.

Reply:
I believe you should remove this test and use repo in previous test to validate user. We want to check if saved user has role updated not if service returned user with given roles.